lib/puppet/util.rb

```diff
@@ -477,11 +477,11 @@ def uri_encode(path, opts = { :allow_fragment => false })
   end
   module_function :uri_encode
 
-  def safe_posix_fork(stdin=$stdin, stdout=$stdout, stderr=$stderr, &block)
+  def safe_posix_fork(stdin=STDIN, stdout=STDOUT, stderr=STDERR, &block)
```

Comment:
I think the defaults should be "wherever the parent process was printing", so these should still be the global variable form, not the constants.